home-application/src/main/java/com/softserveinc/ita/homeproject/application/api/CommonApi.java

```diff
@@ -6,28 +6,34 @@
 
 import com.softserveinc.ita.homeproject.application.mapper.HomeMapper;
 import com.softserveinc.ita.homeproject.homeservice.dto.BaseDto;
-import com.softserveinc.ita.homeproject.model.BaseView;
-import lombok.NoArgsConstructor;
+import com.softserveinc.ita.homeproject.homeservice.query.EntitySpecificationService;
+import com.softserveinc.ita.homeproject.model.BaseReadView;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.data.domain.Page;
+import org.springframework.stereotype.Component;
 
-@NoArgsConstructor
-public abstract class CommonApi<R extends BaseDto> {
+@Component
```

Comment:
Do we really need it here? AFAIK spring cannot instantiate abstract classes as well as plain java. Can we try to remove it?

Reply:
That is right. Removed annotation.